src/linux.rs

```diff
@@ -300,8 +300,17 @@ impl<'a> I2CTransfer<'a> for LinuxI2CDevice {
 
     /// Issue the provided sequence of I2C transactions
     fn transfer(&mut self, messages: &'a mut [Self::Message]) -> Result<u32, LinuxI2CError> {
+        let msg_type = |flag: u16| flag & I2CMessageFlags::READ.bits();
+        let mut prev_msg_type = !msg_type(messages[0].flags);
```

Comment:
I'm not understanding what the `!` is for here. Is it just so the `prev_msg_type` isn't equal to `cur_msg_type` on the first message? If this is the case it'd be much clearer IMO to represent this with an `Option`.

Reply:
you're right. I think my head was stuck in C mode.